Forward keyword arguments when using @deprecate without a signature

danielmatz opened the pull request:
When using the `@deprecate` macro without specifying the deprecated
signature, a method is created for the old name that accepts a
variable number of positional arguments. This PR makes the created
method also accept a variable number of keyword arguments.

This came up with a recent issue with with Interpolations.jl: JuliaMath/Interpolations.jl#507

danielmatz committed:
Forward keyword arguments when using @deprecate without a signature

When using the @deprecate macro without specifying the deprecated
signature, a method is created for the old name that accepts a
variable number of positional arguments. This commit makes the created
method also accept a variable number of keyword arguments.

andyferris reviewed (approved):
Great idea - LGTM!

KristofferC commented:
A compat note should be added that keyword arguments are only forwarded on 1.9.

Also, a test would be good I think.

danielmatz committed:
Test that positional arguments are forwarded to deprecated method

danielmatz commented:
OK, I added a compat and a test.

I also noticed that the old behavior of forwarding positional arguments wasn't being tested, so I added a test for that.
